libglusterfs/src/glusterfs/gf-dirent.h

```diff
@@ -56,8 +56,8 @@ struct _gf_dirent {
 static inline gf_boolean_t
 inode_dir_or_parentdir(gf_dirent_t *entry)
 {
-    return (entry->d_len <= 2 && (strcmp(entry->d_name, ".") == 0 ||
-                                  strcmp(entry->d_name, "..") == 0));
+    return ((entry->d_name[0] == '.') &&
+            ((entry->d_name[1] == 0) || (entry->d_name[1] == '.')));
```

Comment:
Without checking `d_len`, you need to verify that `d_name[2] == 0` in case of `..`.

If you prefer to not use d_len, why not just accept a `char *` instead of a `gf_dirent_t *` ? it's more generic and does the same.

Reply:
Added back the check to entry->d_len